radicle-cli/src/commands/remote/add.rs

```diff
@@ -0,0 +1,52 @@
+use radicle::node::tracking::store::Config;
+use radicle::{git::Url, node::TRACKING_DB_FILE, prelude::Id, Profile};
+use radicle_crypto::PublicKey;
+
+use crate::{git, terminal as term};
+
+pub fn run(
+    repository: &git::Repository,
+    profile: &Profile,
+    pubkey: &PublicKey,
+    name: &Option<String>,
+    id: Id,
+) -> anyhow::Result<()> {
+    let Some(name) = get_remote_name(profile, pubkey, name)? else {
+        anyhow::bail!("an alias need to be specified");
+    };
+
+    if git::is_remote(repository, &name)? {
+        anyhow::bail!("remote with alias `{name}` already exists");
+    }
+    let url = Url::from(id).with_namespace(*pubkey);
+    let url = url.to_string();
+    let remote = repository.remote(&name, &url)?;
+    let (Some(name), Some(url)) = (
+        remote.name(),
+        remote
+            .url()) else {
+        // This else branch should never be hit, if this happens
+        // there is some wired stuff that is going on and we
+        //  try to steal some information for an easy debug!
+        debug_assert!(remote.name() == Some(&name));
+        debug_assert!(remote.url() == Some(&url));
+        anyhow::bail!("BUG: remote filed are not correctly built!");
+    };
+
+    term::success!("Remote {name} added with {url}");
+    Ok(())
+}
+
+/// Get the `git remote` name from the command `Options` and `pubkey`.
+fn get_remote_name(
+    profile: &Profile,
+    pubkey: &PublicKey,
+    name: &Option<String>,
+) -> anyhow::Result<Option<String>> {
+    if name.is_some() {
+        return Ok(name.to_owned());
+    }
```

Comment:
Doesn't really make sense for this check to be inside this function. If `name` is `Some`, it doesn't need to be passed in here at all.

Reply:
I propose a solution with  `map_or`